polyply/src/minimizer.py

```diff
@@ -173,18 +173,25 @@ def target_function(positions):
 
     opt_results = scipy.optimize.minimize(target_function, positions, method='L-BFGS-B',
                                           options={'ftol': 0.001, 'maxiter': 100})
+
+    # evaluate if optimization was successful
     positions = opt_results['x'].reshape((-1, 3))
     positions = renew_vs(positions, block, atom_to_idx)
+
     for node_key, idx in atom_to_idx.items():
         coords[node_key] = positions[idx]
 
-    max_fun = 0
-    for inter_type, tol in tolarance.items():
-        max_fun += tol**2. * len(block.interactions[inter_type])
-
-    if opt_results['success'] and opt_results['fun'] < max_fun:
-        success = True
-    else:
-        success = False
-
-    return success, coords
+    for inter_type in inter_types:
+        interactions = block.interactions.get(inter_type, [])
+        for interaction in interactions:
+            atoms = interaction.atoms
+            params = interaction.parameters
+            atom_coords = [positions[atom_to_idx[name]]
+                           for name in atoms]
+            penalty = INTER_METHODS[inter_type](params, atom_coords)
+            if inter_type in ["bonds", "constraints"] and penalty > 10000*tolarance[inter_type]**2.:
+                return False, coords
+            elif penalty > 10000*tolarance[inter_type]**2.:
```

Comment:
I don't like these hardcoded force constants. (And the hardcoded functional form is debatable, but I can live with that)
Instead, make a global dict with interaction_type -> force constant that is used both here and by the objective function.
Or even bake them into the INTER_METHODS, e.g. using functools.partial

Reply:
So I moved the constants to a global dict WEIGHTS, which is now flexible. But this is something I typically don't want to expose to the user and so I'm more than happy with it being somewhat hard-coded. Depending on the FF one could think to use different weights, but that is very much in the future.